contrib/test_jenkins.sh

```diff
@@ -932,6 +957,8 @@ run_mpi_tests() {
 		# our local library path first
 		export LD_LIBRARY_PATH=${ucx_inst}/lib:$LD_LIBRARY_PATH
 
+		try_load_cuda_env
```

Comment:
add unload_cuda_env in the end of this function?

Reply:
@yosefe 
we decided earlier to load cuda modules at the very beginning of the Jenkins and keeps the module for all the tests/build.  I loaded it hear again just to be sure in case if anybody unloads them by mistake. 

the current approach is to unload them if any specific test does not work when cuda modules are loaded. It seems the following tests on legacy setups failed with cuda builds.  So I unloaded and loaded again after these tests.
```

run_gtest_default
run_gtest_armclang
```

Reply:
then maybe instead of loading them, fail if they are not loaded? this will catch errors

Reply:
also it seems like unloading cuda before "run_gtest_default" means all gtests would be run without cuda?

Reply:
then it will be not generic.  we don't expect them to be available on setups.

Reply:
> then maybe instead of loading them, fail if they are not loaded? this will catch errors

there is another one. what does it do?
` do_distributed_task 1 4 run_gtest_release`

Reply:
i see. but why do we think the could be unloaded "by mistake"?
on another note - maybe load only if NVIDIA GPU is available? this may fix failures on legacy setups?

Reply:
> there is another one. what does it do?
> ` do_distributed_task 1 4 run_gtest_release`

Runs some specific tests in release mode, like structures size check

Reply:
> i see. but why do we think the could be unloaded "by mistake"?
> on another note - maybe load only if NVIDIA GPU is available? this may fix failures on legacy setups?

before this PR, we mistakenly unloaded here: https://github.com/openucx/ucx/blob/master/contrib/test_jenkins.sh#L480

you are right. it is gtest_default in all setups. Will check why it is failing in legacy setups.

Reply:
> before this PR, we mistakenly unloaded here: https://github.com/openucx/ucx/blob/master/contrib/test_jenkins.sh#L480

seems like we should change build_cuda() function so it would not load modules, but instead just pass --with-cuda and --with-gdrcopy if those are available?

Reply:
seems better. will change.
 BTW, why do we need to run test_link_map with out cuda module
```
 482         echo "==== Running test_link_map with cuda build but no cuda module ===="
 483         env UCX_HANDLE_ERRORS=bt ./test/apps/test_link_map
```

Reply:
the intent of this test was to run when one of the UCT modules is missing runtime dependencies so it fails to load. I guess we'll need to wrap it with cuda unload+cuda load..